src/training/validator.h

```diff
@@ -125,13 +126,13 @@ class Validator : public ValidatorBase {
 
 class CrossEntropyValidator : public Validator<data::Corpus> {
 public:
-  CrossEntropyValidator(std::vector<Ptr<Vocab>> vocabs, Ptr<Config> options)
+  CrossEntropyValidator(std::vector<Ptr<Vocab>> vocabs, Ptr<Options> options)
       : Validator(vocabs, options) {
-    Ptr<Options> opts = New<Options>();
-    opts->merge(options);
-    opts->set("inference", true);
-    opts->set("cost-type", "ce-sum");
-    builder_ = models::from_options(opts, models::usage::scoring);
+    // set the cost type temporarily to 'ce-sum' for model builder
+    auto ctype = options_->get<std::string>("cost-type");
+    options_->set("cost-type", "ce-sum");
+    builder_ = models::from_options(options_, models::usage::scoring);
+    options_->set("cost-type", ctype);
   }
```

Comment:
Seems to me like CEValidator should have it's own option object that can be safely modified?

Reply:
It's not because options_ can not be modified here for some reason, but because "cost-type" seems to mean something different for models and for validators? Very confusing, but I haven't investigated this deeper as I assumed to be out of the scope of this PR.